txnkv/transaction/txn.go

```diff
@@ -902,7 +902,8 @@ func (txn *KVTxn) collectLockedKeys() [][]byte {
 	var err error
 	for it := buf.IterWithFlags(nil, nil); it.Valid(); err = it.Next() {
 		_ = err
-		if it.Flags().HasLocked() {
+		flags := it.Flags()
+		if flags.HasLocked() || flags.HasLockedInShareMode() {
```

Comment:
The original `it.Flags().HasLocked()` is ok actually. `flagKeyLockedInShareMode` is used to indicate the lock mode (exclusive or shared), so `flags.HasLocked()` means the key is locked in either exclusive mode or shared mode, ref the comment https://github.com/tikv/client-go/pull/1784/files#diff-a2d641d4e33ddcd918adaa4ead7b0bfc9bcdfca013c06d4defb7a7fb86d283d9R81-R85 .